wrappers/clang-scan-deps-wrapper.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/sh
+#
+# Copyright (c) 2024 Martin Storsjo
+#
+# Permission to use, copy, modify, and/or distribute this software for any
+# purpose with or without fee is hereby granted, provided that the above
+# copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+get_dir() {
+    target="$1"
+    while [ -L "$target" ]; do
+        cd "$(dirname "$target")"
+        target="$(readlink "$(basename "$target")")"
+    done
+    cd "$(dirname "$target")"
+    pwd
+}
+
+DIR="$(get_dir "$0")"
+BASENAME="$(basename "$0")"
+TARGET="${BASENAME%-clang-scan-deps}"
+DEFAULT_TARGET=x86_64-w64-mingw32
+if [ "$TARGET" = "$BASENAME" ]; then
+    TARGET=$DEFAULT_TARGET
+fi
+
+# If changing this wrapper, change clang-scan-deps-wrapper.c accordingly.
+EXE="$DIR/clang-scan-deps"
+FLAGS=""
+while [ $# -gt 0 ]; do
+    if [ -n "$got_flags" ]; then
+        CMD_EXE="$1"
+        shift
+        break
+    elif [ "$1" = "--" ]; then
+        got_flags=1
+        FLAGS="$FLAGS $1"
+    else
+        FLAGS="$FLAGS $1"
+    fi
+    shift
+done
+
+if [ -n "$CMD_EXE" ]; then
+    # If CMD_EXE is a <triple>-<exe> style command, prefer the
+    # target triple from there, rather than from what we might have
+    # had in our name.
+    CMD_BASENAME="$(basename "$CMD_EXE")"
+    CMD_BASENAME="${CMD_BASENAME%.*}"
+    CMD_TARGET="${CMD_BASENAME%-*}"
+    CMD_EXE_SUFFIX="${CMD_BASENAME##*-}"
+    if [ "$CMD_TARGET" != "$CMD_BASENAME" ]; then
+        case $CMD_EXE_SUFFIX in
+        clang|clang++|gcc|g++|c++|as|cc|c99|c11)
+            TARGET="$CMD_TARGET"
```

Comment:
Since `clang` can infer the target triple implicitly, does it make sense to have a upstream fix for `clang-scan-deps` to infer the target triple from `CMD_EXE` ? ( not from `clang-scan-deps` itself ! )

Reply:
Yes, I think it could make sense to try make upstream `clang-scan-deps` infer triples in such a similar way, possibly looking both at `CMD_EXE` and `argv[0]`, like this script does. But that would be a later feature, maybe for LLVM 19 or so, depending on whether others agree with it.

Even if we do that, I fear we can't get rid of our wrapper script, since we need it to set `-stdlib=libc++`.